src/WTG.BulkAnalysis.Core/Internal/AnalyzerCache.cs

```diff
@@ -107,11 +107,20 @@ static ImmutableArray<T> GetCached<T>(string assemblyName, ConcurrentDictionary<
 
 			static IEnumerable<string> GetAnalyzerRefs(Solution solution)
 			{
-				return solution
-					.Projects
-					.SelectMany(p => p.AnalyzerReferences)
-					.Select(a => a.FullPath)
-					.Distinct(StringComparer.OrdinalIgnoreCase);
+				var result = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+
+				foreach (var project in solution.Projects)
+				{
+					foreach (var reference in project.AnalyzerReferences)
+					{
+						if (!string.IsNullOrEmpty(reference.FullPath))

```

Comment:
Is the change here purely for nullability analysis?

Reply:
Yes.